controller/team.go

```diff
@@ -0,0 +1,103 @@
+package controller
+
+import (
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/application"
+	"github.com/fabric8-services/fabric8-auth/authorization"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/jsonapi"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/login"
+	"github.com/goadesign/goa"
+	"strings"
+)
+
+// TeamController implements the team resource.
+type TeamController struct {
+	*goa.Controller
+	app application.Application
+}
+
+// NewTeamController creates a team controller.
+func NewTeamController(service *goa.Service, app application.Application) *TeamController {
+	return &TeamController{Controller: service.NewController("TeamController"), app: app}
+}
+
+// Create runs the create action.
+func (c *TeamController) Create(ctx *app.CreateTeamContext) error {
+	currentUser, err := login.ContextIdentity(ctx)
```

Comment:
Above I asked to check if the user is deprovisioned in TeamService before proceed. But as alternative we could check it in the controller instead. Maybe it's better to do in controller actually. Not sure. In other places we do it in controllers.
In this case the call would be something like this:

```
currentUser, err := login.LoadContextIdentityIfNotDeprovisioned(ctx, c.app)
if err != nil {
	return jsonapi.JSONErrorResponse(ctx, err)
}
```